addons/account/models/chart_template.py

```diff
@@ -499,33 +502,38 @@ def deref(values, model):
                     ])]
             return values
 
-        def defer(all_data):
+        def delay(all_data):
             """Defer writing some relations if the related records don't exist yet."""
+
+            def should_delay(created_models, yet_to_be_created_models, model, field_name, field_val, parent_models=None):
+                parent_models = (parent_models or []) + [model]
+                field = self.env[model]._fields.get(field_name)
+                if not (field and field.relational and field.comodel_name not in created_models):
+                    return False
+                field_yet_to_be_created = field.comodel_name in parent_models + yet_to_be_created_models
+                if not isinstance(field_val, list | tuple):
+                    return field_yet_to_be_created
+                # Check recursively if there are subfields that should be delayed
+                for element in field_val:
+                    match element:
+                        case Command.CREATE, _, dict() as values:
+                            for subkey, subvalue in values.items():
+                                if should_delay(created_models, yet_to_be_created_models, field.comodel_name, subkey, subvalue, parent_models):
+                                    return True
+                        case Command(), *_:
```

Comment:
![image](https://github.com/odoo/odoo/assets/1665365/20866ae7-1545-496a-adc2-6e22f42ed7b3)

Reply:
Now try again with
```python
x = (5, 0, 0)
match x:
    case int() as c, *_ if c in Command:
        print(c)
```

Reply:
Fixin', but this is bonkers. 
![image](https://github.com/odoo/odoo/assets/1665365/ae12f82a-71a0-433f-b3df-967a07daf59d)

We should write a test with numbers though.
I don't think we have one since also `l10n` tests were passing.
I guess the functions can be called for whatever so it's right to support numbers anyway.